### ISSUE TITLE
FFmpeg support

### ISSUE THREAD

**Issue by G-Burnett** [humhub/humhub#6025]
### Describe alternatives you've considered

I am about to use HumHub to give our Year 6 children (10 - 11  yrs old) a relatively safe experience of using social media in a closed environment. Many of them will be using iPhones, iPads etc to upload images and videos.

Whilst the ability to compress the very large images that these devices create is built in to HumHub, there is no such integration for video compression. This means submitted videos take up a large amount of disk space and more importantly playback is poor over slow connections.

Would it be possible to develop a PHP service that could monitor the 'Uploads' folder and detect the submission of a video, on detection of video (.mov .mp4) FFmpeg would then compress the video to a data rate of 2mbps (typically) then add the video to the users post.

If anybody is able to develop this feature, it would be very much appreciated.

**Comment by funkycram** [comment]
Keywords: **video compression**
Should work the same way as https://docs.humhub.org/docs/admin/uploads/#image-compression--downscaling

**Comment by ArchBlood** [comment]
I recommend the following;
https://github.com/PHP-FFMpeg/PHP-FFMpeg

**Comment by G-Burnett** [comment]
> I recommend the following; https://github.com/PHP-FFMpeg/PHP-FFMpeg

This is unfortunately beyond my capabilities to implement:-(

**Comment by tpnsolutions** [comment]
I agree, post processing of a video to reduce file size and resolution would be helpful. A lot of our users don't know how to do this on their end before uploading.

**Comment by G-Burnett** [comment]
Is there anybody out there keen to develop a module to provide this increasingly necessary functionality? It would be very welcome indeed.

**Comment by CYBERNEURONES** [comment]
Hi,
  Currently I try to find workaround like : 
```
# find /var/www/humhub/uploads/file/ -name 'file' -exec file {} + | grep "Apple QuickTime movie" | sed 's/:/ /g' | awk '{print $1}' | xargs du -b | awk '{ sum += $1 } END { print sum/1024/1024 }'
978.387
# find /var/www/humhub/uploads/file/ -name 'file' -exec file {} + | grep "MP4 " | sed 's/:/ /g' | awk '{print $1}' | xargs du -b | awk '{ sum += $1 } END { print sum/1024/1024 }'
6690.92
# find /var/www/humhub/uploads/file/ -name 'file' -exec file {} + | grep "MP4 " | sed 's/:/ /g' | awk '{print $1}' | xargs -I {} ffmpeg -i {} -c:v libx264 -preset slow -crf 18 -c:a aac -vf format=yuv420p -movflags +faststart   {}_tmp7.mp4
# find /var/www/humhub/uploads/file/ -name 'file_tmp7.mp4' -exec file {} + | grep "MP4 " | sed 's/:/ /g' | awk '{print $1}' | sed 's/_tmp7.mp4//g' | xargs -I {} mv {}_tmp7.mp4 {}
# find /var/www/humhub/uploads/file/ -name 'file' -exec file {} + | grep "MP4 " | sed 's/:/ /g' | awk '{print $1}' | xargs du -b | awk '{ sum += $1 } END { print sum/1024/1024 }'
4306.44
# chown -R www-data.www-data  /var/www/humhub
# php yii integrity/run
# php yii cache/flush-all
```

How to update size ?

<img width="316" height="122" alt="Image" src="https://github.com/user-attachments/assets/130cdafb-6786-4e4e-bce1-4d7885cd568f" />

Thanks.
